### PR TITLE
Rename p256m configuration option

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Fixes https://github.com/Mbed-TLS/mbedtls/issues/8202

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** N/A
- [x] **backport** N/A
- [x] **tests** N/A

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

File: docs/psa-driver-example-and-guide.md
Comment (tom-cosgrove-arm) [inline docs/psa-driver-example-and-guide.md:149]: This says "You need extra steps if you want to disable the built-in implementation of ECC algorithms" and "Refer to the documentation of `MBEDTLS_PSA_P256M_DRIVER_ENABLED` for more information" but I don't see those extra steps in `mbedtls_config.h`

File: library/psa_crypto_driver_wrappers.h
Comment (tom-cosgrove-arm) [inline library/psa_crypto_driver_wrappers.h:29]: (pre-existing) probably could just remove the comment on this line; would improve readability

File: docs/psa-driver-example-and-guide.md
Comment (gilles-peskine-arm) [reply docs/psa-driver-example-and-guide.md:149]: (preexisting, but ok, as long as this doesn't take too much back-and-forth) The steps are there, but it's true that they don't explicitly state that they let you “disable the built-in implementation”, they just explain what you can do without enabling the built-in implementation. I guess more verbiage won't hurt so I added another explanation that takes things the other way round.

File: library/psa_crypto_driver_wrappers.h
Comment (gilles-peskine-arm) [reply library/psa_crypto_driver_wrappers.h:29]: Sorry, too out of scope and unimportant.

File: docs/psa-driver-example-and-guide.md
Comment (mpg) [reply docs/psa-driver-example-and-guide.md:149]: Note: `docs/driver-only-builds.md` is supposed to have all the details on how to build with a driver and without the built-in implementation in general. Ideally, we'd reference from the documentation of `MBEDTLS_PSA_P256M_DRIVER_ENABLED`, which would only need to explain the bits that are specific to this particular driver.

*If you find something is missing, please shout and I'll fix it as part of #7757.

Comment (tom-cosgrove-arm) [reply docs/psa-driver-example-and-guide.md:149]: @mpg Are you reviewing this PR, or do we still need a second reviewer?

Comment (mpg) [reply docs/psa-driver-example-and-guide.md:149]: I'm not reviewing, just had a quick look out of curiosity.

Comment (mpg) [reply docs/psa-driver-example-and-guide.md:149]: (More precisely: don't count on me until I'm done with 8075 and 7757 - and by the time I'm finished with those, I hope this will have a 2nd reviewer already.)

Comment (tom-daubney-arm) [reply docs/psa-driver-example-and-guide.md:149]: I will take second review now

**Review by tom-daubney-arm — APPROVED** [review]
LGTM.

File: docs/psa-driver-example-and-guide.md
Comment (gilles-peskine-arm) [reply docs/psa-driver-example-and-guide.md:149]: I'm going ahead and merging. Further documentation improvements are welcome but they aren't release-critical.

**Pull request merged**
